lib/bap_byteweight/bap_byteweight_signatures.ml

```diff
@@ -109,9 +109,8 @@ let update_or_fail ?compiler target data payload path =
   let path = make_entry ?compiler target data in
   let data = Bytes.unsafe_to_string (data.save payload) in
   Zip.add_entry data zip path;
-  List.iter entries ~f:(fun ({Zip.filename; extra; comment; mtime},data) ->
-      Zip.add_entry data zip filename
-        ~extra ~comment ~mtime)
+  List.iter entries ~f:(fun ({Zip.filename; comment; mtime},data) ->
```

Comment:
If you write the pattern as follows, you'll remain compatible with CamlZip < 1.12 while not triggering warning 9:
```
fun ({Zip.filename; comment; mtime; _},data) ->
```